docs/rfc/rfc-7.adoc

```diff
@@ -0,0 +1,237 @@
+
+The signing starts with the announcement phase where each signing group member
+announces their readiness to participate in the signing of sweep transaction inputs.
+51 signers from the ones who announced their readiness are selected for the
+protocol. They unlock each deposit UTXO with a signature. Signatures are
+computed in parallel and no signature should be completed without all others.
+This approach simplifies defeating fraud proofs as this is all-or-nothing
+situation for a sweep transaction signing. There is a timeout for unlocking all
+inputs of the sweep transactions. If the timeout is exceeded, the retry is
+executed. The retry starts with a new announcement phase to establish a set of
+signers ready to execute the protocol.
+
+```
+|                  | ------------ signature 1 ------------> |                   |
+| ... announce ... | ------------ signature 2 ------------> | protocol end      | timeout
+|                  | ------------ signature 3 ------------> |                   |
+```
+
+==== Retries
+
+The most computationally expensive part of the signing protocol takes 2 seconds
+on a 10-core M1 machine for a single signing group member. Assuming the client
+works on a 2-core machine and there are no more than 5 signing group members on
+a single machine, the bottleneck phase should take no more than 1 minute. For
+the sake of establishing a sufficient timeout, we assume a single signing should
+take no more than 5 minutes not counting the announcement phase. It means that
+all signatures for a sweep transaction with a maximum number of 10 inputs
+should be completed in less than an hour.
+
+The timeout for signing should be proportional to the number of inputs in the
+sweep transaction.
+
+The signing protocol is non-attributable and we need to retry in case of
+corrupted data:
+
+- With 2 malicious members in a signing group, we need 5 retries of the protocol
+in the worst case (`P = (98 choose 51) / (100 choose 51) = 0.23757575757`),
+which takes 5 hours for 10 inputs.
+- With 3 malicious members in a signing group, we need 10 retries of the
+protocol in the worst case (`P = (97 choose 51) / (100 choose 51) = 0.11393939393`),
+which takes 10 hours for 10 inputs.
+- With 4 malicious members in a signing group, we need 20 retries of the
+protocol in the worst case (`P = (96 choose 51) / (100 choose 51) = 0.05403311465`),
+which takes 20 hours for 10 inputs.
+
+The signing that keeps failing is retried for no more than 48 hours. Since no
+interim signatures were produced, it is safe to stop the signing without exposing
+wallet for risk of being slashed for signature fraud.
+
+No state management logic is required for retries other than keeping information
+in-memory about the sweep transaction being signed. If more than t+1 clients get
+restarted during the sweep transaction signing, the sweep attempt is skipped and
+everyone needs to wait for another sweep window.
+
+==== Increasing fee
```

Comment:
Worth mentioning that the wallet members need to hold the preimages of all transactions that were not mined in order to be able to defeat potential fraud challenges.